libs/postgres_ffi/src/xlog_utils.rs

```diff
@@ -154,18 +156,23 @@ mod timestamp_conversions {
         }
     }
 
+    /// TODO: this function should be deprecated, likely malicious input could cause the .unwrap() to panic.
```

Comment:
From a grep it seems like the only user was in routes.rs where we could easily return 500 on a failure, so maybe worth fixing inline in this PR?

Up to you.